streams/src/test/java/org/apache/kafka/streams/integration/KTableKTableForeignKeyJoinIntegrationTest.java

```diff
@@ -27,673 +25,314 @@
-        final KeyValueIterator<Integer, String> allTwo = myJoinStoreTwo.all();
-        final KeyValueIterator<Integer, String> allThree = myJoinStoreThree.all();
-
-        final List<KeyValue<Integer, String>> all = new LinkedList<>();
-
-        while (allOne.hasNext()) {
-            all.add(allOne.next());
-        }
-        while (allTwo.hasNext()) {
-            all.add(allTwo.next());
-        }
-        while (allThree.hasNext()) {
-            all.add(allThree.next());
-        }
-        allOne.close();
-        allTwo.close();
-        allThree.close();
-
-        for (final KeyValue<Integer, String> elem : all) {
-            assertTrue(expectedResult.contains(elem));
-        }
+    private static Map<Integer, String> asMap(final KeyValueStore<Integer, String> store) {
+        final HashMap<Integer, String> result = new HashMap<>();
+        store.all().forEachRemaining(kv -> result.put(kv.key, kv.value));
+        return result;
     }
 
-    private KafkaStreams prepareTopology(final String queryableStoreName, final boolean leftJoin) {
+    private static Topology getTopology(final Properties streamsConfig,
+                                        final String queryableStoreName,
+                                        final boolean leftJoin) {
         final StreamsBuilder builder = new StreamsBuilder();
 
         final KTable<Integer, Float> left = builder.table(LEFT_TABLE, Consumed.with(Serdes.Integer(), Serdes.Float()));
         final KTable<String, Long> right = builder.table(RIGHT_TABLE, Consumed.with(Serdes.String(), Serdes.Long()));
 
-        final Materialized<Integer, String, KeyValueStore<Bytes, byte[]>> materialized;
-        if (queryableStoreName != null) {
-            materialized = Materialized.<Integer, String, KeyValueStore<Bytes, byte[]>>as(queryableStoreName)
-                    .withKeySerde(Serdes.Integer())
-                    .withValueSerde(Serdes.String())
-                    .withCachingDisabled();
-        } else {
-            throw new RuntimeException("Current implementation of join on foreign key requires a materialized store");
-        }
+        final Materialized<Integer, String, KeyValueStore<Bytes, byte[]>> materialized =
+            Materialized.<Integer, String>as(Stores.inMemoryKeyValueStore(queryableStoreName))
+                .withKeySerde(Serdes.Integer())
+                .withValueSerde(Serdes.String())
+                .withCachingDisabled();
+
+        final Function<Float, String> extractor = value -> Integer.toString((int) value.floatValue());
+        final ValueJoiner<Float, Long, String> joiner = (value1, value2) -> "value1=" + value1 + ",value2=" + value2;
 
         if (leftJoin)
-            left.leftJoin(right, tableOneKeyExtractor, joiner, Named.as("customName"), materialized)
+            left.leftJoin(right,
+                          extractor,
+                          joiner,
+                          Named.as("customName"),
```

Comment:
Do we need to name it?

Reply:
not sure why this was there.

Reply:
I prefer to avoid noise -- if we don't need to name it, don't name it :) -- but also fine to leave in.

Reply:
Ok, I'll take it out.